### ISSUE TITLE
Error on Slug.js

### ISSUE THREAD

**Issue by pradapdp7** [adrianhajdin/ecommerce_sanity_stripe#20]
![Screenshot (83)](https://user-images.githubusercontent.com/77932658/168413173-0570bf16-593d-4af8-aa3a-52f4612d6a86.png)
  
Hi, I'm getting this issue after running slug.js. I'm currently at 1:28:00min in the video. Kindly  Help me to fix this error

**Comment by miahthegreat** [comment]
The error you are getting appears to be stemming from the Products.jsx file within the components folder, not the [slug].js file. What does your Products.jsx look like?

**Comment by Pannu786** [comment]
Yo @miahthegreat  could you please help me with my error. That would be a huge help. Thank you.

**Comment by miahthegreat** [comment]
@Pannu786 What error are you getting?

**Comment by Pannu786** [comment]
Hey @miahthegreat 
I am stuck at 2:11. I just couldn't figure out why I am getting this error "Unhandled Runtime Error" ..
![Screenshot 2022-05-17 221843](https://user-images.githubusercontent.com/43903886/168981291-5ee2cfb5-4d55-4d2a-b93b-12d566fd3047.png)

![Screenshot 2022-05-17 221107](https://user-images.githubusercontent.com/43903886/168981141-1ec486d2-d23e-467c-88dd-443bb69e3c2b.png)

Cheers

**Comment by pradapdp7** [comment]
> @pradapdp7
> 
> The error you are getting appears to be stemming from the Product.jsx file within the components folder, not the [slug].js file. What does your Product.jsx look like?
![Screenshot (84)](https://user-images.githubusercontent.com/77932658/169698417-f615eff4-21d5-467e-853a-7874c7de1570.png)
this is my product.jsx file sir

**Comment by pradapdp7** [comment]
> @pradapdp7
> 
> The error you are getting appears to be stemming from the Product.jsx file within the components folder, not the [slug].js file. What does your Product.jsx look like?

here, the home page works fine but when i click any product this error pops up

**Comment by sodapopjam** [comment]
No, the error is on slug page. I can't see above params: on the first screenshot, but it says you're missing a ,
**Make sure the line above params on [slug].js ends with a comma ,**

**Comment by pradapdp7** [comment]
> No, the error is on slug page. I can't see above params: on the first screenshot, but it says you're missing a , **Make sure the line above params on [slug].js ends with a comma ,**
![Screenshot (89)](https://user-images.githubusercontent.com/77932658/170246595-b1e13a57-a2d7-49da-8b45-878a3d205405.png)

this is my code above params. can you tell me what to change

**Comment by sodapopjam** [comment]
![170246595-b1e13a57-a2d7-49da-8b45-878a3d205405](https://user-images.githubusercontent.com/69822895/170259980-b55e3e56-a063-4215-8f54-3ff538133cc6.png)
You are missing double and single quotes on the query. Line 23 should say:

`const query = `*[_type == “product”]` {`

**Comment by pradapdp7** [comment]
> _Line 23 should say:_ https://codefile.io/f/O2jutdM2wO62ZiApBPcT
> 
> ![170246595-b1e13a57-a2d7-49da-8b45-878a3d205405](https://user-images.githubusercontent.com/69822895/170260368-c2ec91ae-5935-451c-8c50-09
![Screenshot (90)](https://user-images.githubusercontent.com/77932658/170307976-df1681e3-fc43-4cd6-9c53-8b0f4725ecdc.png)
cf9f8dcdfc.png) {

thankyou so much but after clearing that I got this error.kindly help me :(. Again Thanks in advance:)

**Comment by vaniadimova** [comment]
I am stuck on TypeError: Cannot destructure property 'image' of 'product' as it is null. clicking on the button /// when clicking on the product it works ... my brain got scrambled pls help, pretty please.

export const getStaticPaths = async () => {
  const query = `*[_type == "product"] {
    slug {
      current
    }
  }
  `
const products = await client.fetch(query);
const paths = products.map((product) => ({
    params: { 
      slug: product.slug.current
    }
  }));

  return {
    paths,
    fallback: 'blocking'
  }
}

export const getStaticProps = async ({ params: { slug }}) => {
  const query = `*[_type == "product" && slug.current == '${slug}'][0]`;
  const product = await client.fetch(query);
 
  const productsQuery = '*[_type == "product"]'
  const products = await client.fetch(productsQuery);

  return {
    props: { products, product }
  }
}
